File: clojure/sum-of-multiples/src/sum_of_multiples.clj
(ns sum-of-multiples)

(defn- multiples
  [limit]
  (fn [n]
    (take-while #(< % limit)
                (iterate #(+ % n) n))))

(defn sum-of-multiples
  [nums limit]
  (->> nums
       (mapcat (multiples limit))
       distinct
       (reduce +)))

File: python/sum-of-multiples/sum_of_multiples.py
def sum_of_multiples(limit, multiples):
    all_multiples = set()

    for m in multiples:
        if m == 0:
            all_multiples.add(0)
        else:
            all_multiples.update(range(m, limit, m))

    return sum(all_multiples)
